cmd/kubeadm/app/phases/addons/dns/dns.go

```diff
@@ -41,7 +41,12 @@ const (
 )
 
 // EnsureDNSAddon creates the kube-dns addon
-func EnsureDNSAddon(cfg *kubeadmapi.MasterConfiguration, client clientset.Interface, k8sVersion *version.Version) error {
+func EnsureDNSAddon(cfg *kubeadmapi.MasterConfiguration, client clientset.Interface) error {
+	k8sVersion, err := version.ParseSemantic(cfg.KubernetesVersion)
```

Comment:
I'd prefer to pass `k8sVersion` still and do the parsing client-side in `cmd/phases`
Luckily, I expect to soon get rid of this in the v1.9 cycle by moving this into the internal variant of cfg or something so we can stop passing and parsing this all the time.

Reply:
The only issue I had with this was that I needed all the `EnsureXXXAddon` functions have the same signature. Look at the `EnsureAllAddons` and `runAddonsCmdFunc` functions. I'm happy to do something different if we still prefer to pass `k8sVersion ` in.

Reply:
ok with this for now; will fix it for real later